File: python/7 kyu/324.py
# https://www.codewars.com/kata/582642b1083e12521f0000da/train/python

# Passed

def array_mash(a, b):
    result = []
    for idx in range(len(a)):
        result.extend([a[idx], b[idx]])
        
    return result

output = array_mash([1, 2, 3, 4, 5], ['a', 'b', 'c', 'd', 'e'])
print(output)

File: nodejs/7 kyu/324.js
// https://www.codewars.com/kata/5497a3c181dd7291ce000700/train/javascript

// Passed

function diagonalSum(matrix) {
  let diagonalNumbers = matrix.map((row, idx) => row[idx]);
  let sum = diagonalNumbers.reduce((a, b) => a + b, 0);
  return sum;
}

const output = diagonalSum([
  [1, 2, 3],
  [4, 5, 6],
  [7, 8, 9],
]);
console.log(output);